addons/product/models/product.py

```diff
@@ -410,7 +410,12 @@ def copy(self, default=None):
         For convenience the template is copied instead and its first variant is
         returned.
         """
-        return self.product_tmpl_id.copy(default=default).product_variant_id
+        # copy variant is disabled in https://github.com/odoo/odoo/pull/38303
+        # this returns the first possible combination of variant to make it
+        # works for now, need to be fixed to return product_variant_id if it's
+        # possible in the future
+        tmpl_id = self.product_tmpl_id.copy(default=default)
+        return tmpl_id._create_product_variant(tmpl_id._get_first_possible_combination())
```

Comment:
```suggestion
        template = self.product_tmpl_id.copy(default=default)
        return template.product_variant_id or template._create_product_variant(template._get_first_possible_combination())
```

Avoid using the suffix `_id` for variables storing recordsets ;)
And maybe keep the old behavior when the template already has a variant automatically generated to reduce eventual side-effects ?

Reply:
Yes! that's more robust!